backend/controllers/dpsUserValidation.ts

```diff
@@ -0,0 +1,61 @@
+import { validateEmailFormat } from './userValidation'
+
+const isAlphaString = (str: string) => str.match(/^[A-Za-z]+$/)
+const isAlphaStringOrApostrophe = (str: string) => str.match(/^[A-Za-z']+$/)
+
+const validateDpsUserCreate = (
+  username: string,
+  email: string,
+  firstName: string,
+  lastName: string,
+  defaultCaseloadId: string,
+  validateCaseload: boolean,
+) => {
+  const errors = []
+
+  if (!username) {
+    errors.push({ href: '#username', text: 'Enter a username' })
+  }
+  if (!email) {
+    errors.push({ href: '#email', text: 'Enter an email address' })
+  }
+  if (!firstName) {
+    errors.push({ href: '#firstName', text: 'Enter a first name' })
+  }
+  if (!lastName) {
+    errors.push({ href: '#lastName', text: 'Enter a last name' })
+  }
+  if (validateCaseload && (!defaultCaseloadId || defaultCaseloadId === '--')) {
+    errors.push({ href: '#defaultCaseloadId', text: 'Select a default caseload' })
+  }
+
+  if (errors.length) return errors
+
+  if (username.length < 2) {
+    errors.push({ href: '#username', text: 'Username must be 2 characters or more' })
+  } else if (username.length > 30) {
```

Comment:
Needs to be 35?

Reply:
Needs to be 30 for username - will update